micro-rdk/src/common/credentials_storage.rs

```diff
@@ -88,32 +91,22 @@ impl WifiCredentials {
     }
 }
 
-#[derive(Clone, Debug, Default)]
-pub struct TlsCertificate {
-    pub(crate) certificate: Vec<u8>,
-    pub(crate) private_key: Vec<u8>,
-}
-
-impl From<CertificateResponse> for TlsCertificate {
-    fn from(resp: CertificateResponse) -> Self {
-        Self {
-            certificate: resp.tls_certificate.into_bytes(),
-            private_key: resp.tls_private_key.into_bytes(),
-        }
-    }
-}
-
-pub trait NetworkSettingsStorage {
+pub trait WifiCredentialStorage {
     type Error: Error + Debug + Into<ServerError>;
     fn has_network_settings(&self) -> bool;
     fn store_network_settings(&self, networks: &[NetworkSetting]) -> Result<(), Self::Error>;
     fn get_network_settings(&self) -> Result<Vec<NetworkSetting>, Self::Error>;
     fn reset_network_settings(&self) -> Result<(), Self::Error>;
 
+    // TODO(RSDK-10105): remove the following two after updating template to new apis
     fn has_wifi_credentials(&self) -> bool;
     fn store_wifi_credentials(&self, creds: &WifiCredentials) -> Result<(), Self::Error>;
-    fn get_wifi_credentials(&self) -> Result<WifiCredentials, Self::Error>;
-    fn reset_wifi_credentials(&self) -> Result<(), Self::Error>;
+
+    fn has_default_network(&self) -> bool;
+    fn store_default_network(&self, ssid: &str, password: &str) -> Result<(), Self::Error>;
+    fn get_default_network(&self) -> Result<NetworkSetting, Self::Error>;
+    fn reset_default_network(&self) -> Result<(), Self::Error>;
+    fn get_all_networks(&self) -> Result<Vec<NetworkSetting>, Self::Error>;
```

Comment:
- I get that the old name isn't ideal anymore, but is this renaming more trouble than it is worth right now? Even if you fix the template, it will mean old projects can't upgrade to the next build of micro-rdk without changes.

- If you feel strongly that we should do the renaming, I think you need forwarding functions for all four methods, not just two.

- The forwarding methods should probably be annotated with `deprecated`: https://doc.rust-lang.org/reference/attributes/diagnostics.html#the-deprecated-attribute

Reply:
I had removed the other two as they weren't actively used anywhere, but you're right that they were still available and someone could have been using it in their own project. I will write forwarding functions as well. 

I like the idea of adding the deprecated tag since that will ensure both compatibility and make users aware that they should change it.